include/xsimd/arch/generic/xsimd_generic_details.hpp

```diff
@@ -180,6 +180,20 @@ namespace xsimd
             {
                 return bitwise_cast<int64_t>(self);
             }
+            template <class A>
+            inline batch<float, A> fast_cast(batch<uint32_t, A> const& v, batch<float, A> const&, requires_arch<generic>) noexcept
+            {
+                // see https://stackoverflow.com/questions/34066228/how-to-perform-uint32-float-conversion-with-sse
+                batch<uint32_t, A> msk_lo(0xFFFF);
+                batch<float, A> cnst65536f(65536.0f);
+
+                auto v_lo = batch_cast<int32_t>(v & msk_lo); /* extract the 16 lowest significant bits of self                             */
+                auto v_hi = batch_cast<int32_t>(v >> 16); /* 16 most significant bits of v                                                 */
```

Comment:
I'm a bit confused here; `fast_cast` is supposed to be called by the generic `batch_cast` when a fast cast is available. Replacing the implementation specific to instruction sets with this generic one would end in calling the "slow" batch cast, right? Or am I missing sometihng?

Reply:
Yes I was thinking along the same lines. Can we replace specific implementations with a generic one ?

Reply:
Indeed, I've adjusted the generic conversion to only be selected if we have a fast conversion from int32 to float.